### PR TITLE
libcsptr: init at 2.0.4

### PR DESCRIPTION
**Pull request by fragamus**
###### Motivation for this change

libcsptr package init

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ x ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/libcsptr/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    owner = "Snaipe";
+    repo = "libcsptr";
+    rev = "v${version}";
+    sha256 = "0i1498h2i6zq3fn3zf3iw7glv6brn597165hnibgwccqa8sh3ich";
+  };
+
+  nativeBuildInputs = [ cmake git ];
+
+  meta = with stdenv.lib; {
+    description = "This project is an attempt to bring smart pointer constructs to the (GNU) C programming language.";
```
Comment (pSub) [inline pkgs/development/libraries/libcsptr/default.nix]: Please use a `description` that conforms to https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes.

File: pkgs/development/libraries/libcsptr/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  src = fetchFromGitHub {
+    owner = "Snaipe";
+    repo = "libcsptr";
+    rev = "v${version}";
+    sha256 = "0i1498h2i6zq3fn3zf3iw7glv6brn597165hnibgwccqa8sh3ich";
+  };
+
+  nativeBuildInputs = [ cmake git ];
+
+  meta = with stdenv.lib; {
```
Comment (pSub) [inline pkgs/development/libraries/libcsptr/default.nix]: Please add yourself as a maintainer.

File: pkgs/development/libraries/libcsptr/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  version = "2.0.4";
+
+  src = fetchFromGitHub {
+    owner = "Snaipe";
+    repo = "libcsptr";
+    rev = "v${version}";
+    sha256 = "0i1498h2i6zq3fn3zf3iw7glv6brn597165hnibgwccqa8sh3ich";
+  };
+
+  nativeBuildInputs = [ cmake git ];
```
Comment (pSub) [inline pkgs/development/libraries/libcsptr/default.nix]: Why do you need git as a build input? In my experience this often indicates impurities in the build.

**Review by pSub — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/libcsptr/default.nix (hunk 1 above)
Comment (fragamus) [reply pkgs/development/libraries/libcsptr/default.nix]: done

File: pkgs/development/libraries/libcsptr/default.nix (hunk 2 above)
Comment (fragamus) [reply pkgs/development/libraries/libcsptr/default.nix]: done

File: pkgs/development/libraries/libcsptr/default.nix (hunk 3 above)
Comment (fragamus) [reply pkgs/development/libraries/libcsptr/default.nix]: removed

**Comment by c0bw3b** [comment]
@fragamus could you rebase this and resolve the merge conflict?

**Comment by aanderse** [comment]
@fragamus are you able to continue with this PR?

**Review by pSub — APPROVED** [review]

**Pull request merged**